src/components/dialogs/generator-creation-dialog.js

```diff
@@ -460,86 +489,76 @@ const GeneratorCreationDialog = ({
                         </Grid>
-                        <Grid container spacing={2}>
-                            <Grid item xs={12}>
-                                <h3 className={classes.h3}>
-                                    <FormattedMessage id="Limits" />
-                                </h3>
-                            </Grid>
-                        </Grid>
+
+                        {/* Limits part */}
+                        <GridSection title="Limits" />
+                        <GridSubSection title="ActiveLimits" />
                         <Grid container spacing={2}>
                             {gridItem(minimumActivePowerField, 4)}
                             {gridItem(maximumActivePowerField, 4)}
                             {gridItem(ratedNominalPowerField, 4)}
-                            <Box sx={{ width: '100%' }} />
-                            {gridItem(isReactiveCapabilityCurveOnField, 8)}
-                            <Box sx={{ width: '100%' }} />
-                            {rCCurveError && (
-                                <div className={classes.rccError}>
-                                    <FormattedMessage id="ReactiveCapabilityCurveCreationError" />
-                                </div>
-                            )}
-                            <Box sx={{ width: '100%' }} />
+                        </Grid>
+                        <GridSubSection title="ReactiveLimits" />
+                        <Grid container spacing={2}>
+                            {gridItem(isReactiveCapabilityCurveOnField, 12)}
                             {!isReactiveCapabilityCurveOn &&
                                 gridItem(minimumReactivePowerField, 4)}
                             {!isReactiveCapabilityCurveOn &&
                                 gridItem(maximumReactivePowerField, 4)}
-                            {isReactiveCapabilityCurveOn &&
-                                reactiveCapabilityCurveField}
-                        </Grid>
-                        <Grid container spacing={2}>
-                            <Grid item xs={12}>
-                                <h3 className={classes.h3}>
-                                    <FormattedMessage id="Setpoints" />
-                                </h3>
-                            </Grid>
+
+                            {!!isReactiveCapabilityCurveOn &&
+                                rCCurveError.length > 0 && (
+                                    <Grid container spacing={2}>
+                                        <Grid item xs={12}>
+                                            {rCCurveError.map((value) => (
+                                                <div
+                                                    className={classes.rccError}
+                                                >
+                                                    <FormattedMessage
+                                                        id={value}
```

Comment:
it's not a value, maybe rename this as `messageDescriptorId` then `rCCurveError` is an array of `messageDescriptorId` as well

Reply:
Ok, changed "value" to "messageDescriptorId".
rCCurveError is now reactiveCapabilityCurveErrors.
Also changed the CSS class name to be more generic.